tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSXPathSelectorTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.tests.integration.amqp;
```

Comment:
That's a fair point, and I thought about that while I was writing the test. However, it's worth noting that there are lots of tests in this package that test more than just AMQP (e.g. `org.apache.activemq.artemis.tests.integration.amqp.JMSSelectorTest` which was the basis for this test). I'm not saying that's a good idea necessarily, but it is consistent at this point and I think if we wanted to refactor this we should do it all at one time.

Reply:
I think it would make more sense to put this new test where it actually best fits now, rather than some worse place simply because something else may have done it already. A followup improvement would be fine I guess if it happened, though seems like that could just improve the existing stuff rather than us needing to add yet more stuff it has to improve.

This basically seems like the choice between making things incrementally better now as we do things, or making it incrementally worse 'because someone else might already have' and adding to future cleanup, the type of thing that rarely ever tends to happen. That doesn't seem like much of a choice at all to me.

Reply:
This test extends `org.apache.activemq.artemis.tests.integration.amqp.JMSClientTestSupport` which in turn extends `org.apache.activemq.artemis.tests.integration.amqp.AmqpClientTestSupport` which in turn extends `org.apache.activemq.artemis.tests.integration.amqp.AmqpTestSupport`. Therefore either all these would need to be visible to wherever I move this test or they would need to be refactored and their multi-protocol functionality extracted into something more generic. I'd like to avoid pulling that thread in this PR, although I think it would be a great idea to address multi-protocol JMS tests generally in another PR.

Reply:
I believe other test classes are using them from different packages already, which should mean they are visible.

Reply:
All the aforementioned parent classes are in `org.apache.activemq.artemis.tests.integration.amqp` together so moving `org.apache.activemq.artemis.tests.integration.amqp.JMSXPathSelectorTest` out of `org.apache.activemq.artemis.tests.integration.amqp` and into something more generic like `org.apache.activemq.artemis.tests.integration.multiprotocol` would require refactoring those parent classes. Again, I think refactoring these is a good idea ultimately, just not in this PR.

Reply:
I disagree it would need any significant refactoring of the parents to move the test class, as I dont think the parents need moved at all for the very small thing I am suggesting of putting the class in the package most appropriate for it. Not every parent needs to be in the same package, they often arent.

EVerything in the multiprotocol package is already parented by [differnt] classes outwith that package. I dont think theres a requirement they all share the same parent, but that could certainly be done if there was a desire and sufficient benefit. I would agree that would come later as its a very distinct task, separate from my comments.

Looking at moving e.g. the existing selector test class now, it looks like it might only need either adding the word public to the ConnectionSupplier interface definition, or maybe creating its own file...or even just removing it since we could probably use java.util.function.Supplier. Presumably the classes already in other packages dont use that interface.

Reply:
> ...I dont think the parents need moved at all for the very small thing I am suggesting of putting the class in the package most appropriate for it.

I guess I misunderstood. I thought you wanted the test to move away from the `org.apache.activemq.artemis.tests.integration.amqp` package since it wasn't specifically related to AMQP. Given that, it didn't (and still doesn't) make sense to me for a non-AMQP-specific test to have a parent that is AMQP-specific. That's why I concluded that the generic multi-protocol functionality would have to be refactored out of the AMQP-specific test parent classes.

> Not every parent needs to be in the same package, they often arent.

Agreed. However, to have a generic test with a more specific parent doesn't make a lot of sense to me.

> Everything in the multiprotocol package is already parented by [different] classes outwith that package.

Which package are you referring to specifically here?

> Looking at moving e.g. the existing selector test class now...

Can you clarify which package this test would move to? Is it logically/heirarchically "above" `org.apache.activemq.artemis.tests.integration.amqp`?


To be frank, this is starting to feel a bit like bike shedding to me.